lib/table_of_contents/parser.rb

```diff
@@ -121,6 +124,14 @@ def toc_headings_within(class_name)
       def ul_attributes
         @ul_attributes ||= @configuration.sublist_class.empty? ? '' : %( class="#{@configuration.sublist_class}")
       end
+
+      def use_ordered_list?
+        @configuration.use_ordered_list
+      end
+
+      def list_tag
+        use_ordered_list? ? ORDERED_LIST_HTML_TAG : UNORDERED_LIST_HTML_TAG
```

Comment:
Please Memoize the result.

Also, I feel using constants and separating method is a bit redundant.
I'd say the following code is enough.

```suggestion
        @list_tag ||= @configuration.use_ordered_list? ? 'ol' : 'ul'
```